src/test/java/picard/sam/markduplicates/util/OpticalDuplicateFinderTest.java

```diff
@@ -125,6 +125,54 @@ public void testKeeperAtEndWithinCliqueOfAllOpticalDuplicates() {
         assertEquals(finder.findOpticalDuplicates(locs, locs.get(2)), new boolean[] {true, true, false});
     }
 
+    @Test
+    public void testKeeperOrderDependence() {
+        final Log log = Log.getInstance(OpticalDuplicateFinderTest.class);
+        final OpticalDuplicateFinder finder = new OpticalDuplicateFinder(OpticalDuplicateFinder.DEFAULT_READ_NAME_REGEX, 100, log);
+        List<PhysicalLocation> locs = Arrays.asList(
+                loc(1, 100, 190),
+                loc(1, 100, 280),
+                loc(1,  100, 370),
+                loc(1,  100, 460)
+        );
+        List<PhysicalLocation> locsReordered = Arrays.asList(
+                loc(1, 100, 190),
+                loc(1, 100, 460),
+                loc(1,  100, 370),
+                loc(1,  100, 280)
+        );
+
+        Assert.assertEquals(countTrue(finder.findOpticalDuplicates(locs, locs.get(0))),3);
+        Assert.assertEquals(countTrue(finder.findOpticalDuplicates(locsReordered, locsReordered.get(0))), 3);
+    }
+
+    @Test
+    public void testKeeperOrderDependenceTwoDistinctClusters() {
+        final Log log = Log.getInstance(OpticalDuplicateFinderTest.class);
```

Comment:
The API supports a null `log` so there's no need to make one for a test, unless the test is testing the log output.